clients/typescript/src/drivers/capacitor-sqlite/adapter.ts

```diff
@@ -0,0 +1,153 @@
+import { capSQLiteSet } from '@capacitor-community/sqlite'
+import {
+  DatabaseAdapter as DatabaseAdapterInterface,
+  RunResult,
+  TableNameImpl,
+  Transaction as Tx,
+} from '../../electric/adapter'
+import { Row, SqlValue, Statement } from '../../util'
+import { Database } from './database'
+import { Mutex } from 'async-mutex'
+
+export class DatabaseAdapter
+  extends TableNameImpl
+  implements DatabaseAdapterInterface
+{
+  #txMutex: Mutex
+
+  constructor(public db: Database) {
+    super()
+    this.#txMutex = new Mutex()
+  }
+
+  async run({ sql, args }: Statement): Promise<RunResult> {
+    if (args && !Array.isArray(args)) {
+      throw new Error(
+        `capacitor-sqlite doesn't support named query parameters, use positional parameters instead`
+      )
+    }
+
+    const wrapInTransaction = false // Default is true. electric calls run from within transaction<T> so we need to disable transactions here.
+
+    const result = await this.db.run(sql, args, wrapInTransaction)
+    const rowsAffected = result.changes?.changes ?? 0
+    return { rowsAffected }
+  }
+
+  async runInTransaction(...statements: Statement[]): Promise<RunResult> {
+    if (statements.some((x) => x.args && !Array.isArray(x.args))) {
+      throw new Error(
+        `capacitor-sqlite doesn't support named query parameters, use positional parameters instead`
+      )
+    }
+
+    const set: capSQLiteSet[] = statements.map(({ sql, args }) => ({
+      statement: sql,
+      values: args as SqlValue[] | undefined,
```

Comment:
Capacitor-sqlite expects an empty array when there are no bound values:

```suggestion
      values: args as SqlValue[] || [],
```

Reply:
> Capacitor-sqlite expects an empty array when there are no bound values:

No need to add the empty array type, `SqlValue[]` also encapsulates the empty array :-)

Reply:
I believe the issue is deeper here. capacitor-sqlite expects an empty array when there are no bind parameters, so we should not cast it to the desired type but we should instead make sure it is never undefined. So:
`values: args ?? []`

Reply:
Ok, i see, we can cast it then 👍

Reply:
Fixed in 8a301ac51d9b81eb828f77827fda5a2ee67c3e66

Reply:
@samwillis sorry i read your comment in a hurry and clearly was confused. You are completely right.